Data/Aeson.hs

```diff
@@ -220,6 +224,15 @@ eitherDecodeStrict' =
   eitherFormatError . eitherDecodeStrictWith jsonEOF' ifromJSON
 {-# INLINE eitherDecodeStrict' #-}
 
+-- | Decode a nested JSON-encoded string.
+embeddedJSON :: (FromJSON a) => String -> (Value -> Parser a)
+embeddedJSON str = withText str (either fail return . eitherDecode . fromStrict . T.encodeUtf8)
+  where
+#if MIN_VERSION_bytestring(0, 9, 2)
```

Comment:
`L.fromChunks . (:[])` (or `L.fromChunks . pure`) works for all versions of bytestring so there is no need for CPP.

Reply:
But `L.fromStrict` doesn't and it's more complex than just `L.fromChunks . (:[])`. So the code should use it except when it's unavailable.

Reply:
Do you mean it's more performant? They should both be O(1) since there is only one chunk, right? Is there a measured performance improvement for it (doesn't have to be within aeson)?

I'd like to avoid CPP, but if we keep it please put it in an unexposed `Data.Aeson.Compat` module,  I've wanted to move most our CPP in there.

Reply:
https://hackage.haskell.org/package/bytestring-0.10.0.2/docs/src/Data-ByteString-Lazy.html#toStrict

The version of `fromStrict` that ByteString provides is performance-optimized, yes. It uses no extra space beyond the output bytestring and copies the data over with `memcpy()`.

`Data.Aeson.Compat` sounds like a decent compromise. Literally the only reason we're having this discussion is because of GHC 7.4.2.